src/main/java/org/broadinstitute/hellbender/utils/nio/SeekableByteChannelPrefetcher.java

```diff
@@ -227,17 +248,38 @@ public int read(ByteBuffer dst) throws IOException {
             src = fetch(position);
             msWaitingForData += waitingForData.elapsed(TimeUnit.MILLISECONDS);
         } catch (InterruptedException e) {
+            // Restore interrupted status
+            Thread.currentThread().interrupt();
```

Comment:
The interrupt status was reset when the exception was thrown (it's there to indicate someone's trying to kill this thread). We cannot add a checked exception to the signature since this method is overridden, so instead we return as soon as possible and ask the caller to not block (calling interrupt() will cause most blocking functions to throw instead). This brings us closer to the goal, which is to unwind the thread's whole stack so it can be interrupted safely.